pkg/cloudprovider/providers/aws/aws_loadbalancer_test.go

```diff
@@ -222,3 +224,121 @@ func TestSecurityGroupFiltering(t *testing.T) {
+				{InstancePort: aws.Int64(80), InstanceProtocol: aws.String("TCP"), LoadBalancerPort: aws.Int64(80), Protocol: aws.String("TCP"), SSLCertificateId: aws.String("def-456")},
+			},
+		},
+		{
+			loadBalancerName: "lb_two",
+			listeners: []*elb.Listener{
+				{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP"), SSLCertificateId: aws.String("abc-123")},
+				{InstancePort: aws.Int64(80), InstanceProtocol: aws.String("TCP"), LoadBalancerPort: aws.Int64(80), Protocol: aws.String("TCP"), SSLCertificateId: aws.String("def-456")},
+			},
+			listenerDescriptions: []*elb.ListenerDescription{
+				{Listener: &elb.Listener{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP"), SSLCertificateId: aws.String("abc-123")}},
+			},
+			toCreate: []*elb.Listener{
+				{InstancePort: aws.Int64(80), InstanceProtocol: aws.String("TCP"), LoadBalancerPort: aws.Int64(80), Protocol: aws.String("TCP"), SSLCertificateId: aws.String("def-456")},
+			},
+		},
+		{
+			loadBalancerName: "lb_three",
+			listeners: []*elb.Listener{
+				{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP"), SSLCertificateId: aws.String("abc-123")},
+			},
+			listenerDescriptions: []*elb.ListenerDescription{
+				{Listener: &elb.Listener{InstancePort: aws.Int64(80), InstanceProtocol: aws.String("TCP"), LoadBalancerPort: aws.Int64(80), Protocol: aws.String("TCP")}},
+				{Listener: &elb.Listener{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP"), SSLCertificateId: aws.String("abc-123")}},
+			},
+			toDelete: []*int64{
+				aws.Int64(80),
+			},
+		},
+		{
+			loadBalancerName: "lb_four",
+			listeners: []*elb.Listener{
+				{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP")},
+			},
+			listenerDescriptions: []*elb.ListenerDescription{
+				{Listener: &elb.Listener{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP"), SSLCertificateId: aws.String("abc-123")}},
+				{Listener: nil},
+			},
+			toDelete: []*int64{
+				aws.Int64(443),
+			},
+			toCreate: []*elb.Listener{
+				{InstancePort: aws.Int64(443), InstanceProtocol: aws.String("HTTP"), LoadBalancerPort: aws.Int64(443), Protocol: aws.String("HTTP")},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		createRequest, deleteRequest := syncElbListeners(test.loadBalancerName, test.listeners, test.listenerDescriptions)
+		assert.Equal(t, createRequest.Listeners, test.toCreate)
+		assert.Equal(t, deleteRequest.LoadBalancerPorts, test.toDelete)
+	}
+}
+
+func TestElbListenersAreEqual(t *testing.T) {
+	tests := []struct {
+		expected, actual *elb.Listener
```

Comment:
can we add a name field to add a description for each test case?

Reply:
done